src/components/earn/FarmBanner/FarmBanner.types.ts

```diff
@@ -0,0 +1,37 @@
+import { ChefVersions } from '../../../state/stake/stake-constants'
+import { Token, TokenAmount } from '@trisolaris/sdk'
+
+export type LpTokenProps = {
+  lpAddress: string
+  token0: Token
+  token1: Token
+}
+
+type Farm = {
+  farmId: number
+  userLPStakedAmount: TokenAmount | null
+  totalStakedInPool: TokenAmount
+  totalPoolAmountUSD: number
+  chefVersion: ChefVersions
+  lpToken: Token
+}
+
+export type DummyCompProps = {
+  farm: Farm
+  updateAmounts: (
+    id: number,
+    stakedAmount: number,
+    rewards: number,
+    parsedNewDoubleRewardsAmount: number,
+    doubleRewardsToken: string
+  ) => void
+}
+
+export type FarmAmount = {
+  [id: number]: {
+    totalStaked: number
+    tokens: {
+      [id: string]: number
+    }
+  }
+}
```

Comment:
Is there a reason this needs to be `key`'d? If these items were pushed as objects in to an array, could the portfolio balance show the sum of all items in the array

It'd make managing a flat data structure easier, too